SU-HSDO/suhumsci
# STN-601: Raised cards modifier/utility class for uniform height

# READY FOR REVIEW

## Summary
This work creates a modifier for `hb-raised-cards` that can be used as a utility class for those views and paragraph layouts within layout builder and the field display.  That utility class `hb-raised-cards--uniform-height` makes all the raised cards in a paragraph row or views grid have a uniform height.

One of the challenges that came from applying this styling was width also not being consistent. This work should also address those issues with the application.

Jira Card: [STN-601](https://sparkbox.atlassian.net/browse/STN-601)

## Need Review By (Date)
10/17

## Urgency
medium

## Steps to Test
1. npm run test
2. To test on Drupal Views card gird layout, go to: http://sparkbox-sandbox.suhumsci.loc/people/people-grid
- Edit the view, using the contextual link.
- Add the classes `hb-raised-cards` `hb-raised-cards--uniform-height` to the "CSS Class" area under Other on the bottom right side of the Views settings.
- Save the View
3. To test on the paragraphs layout using field display layout, go to: http://sparkbox-sandbox.suhumsci.loc/qa/qa-raised-postcards
- Scroll down to the first set of 3 vertical cards.
- Click the contextual link in the top right of the last card that says "manage display".
- Click on the gear on the right side to edit the "components" field display settings.
- Add the class`hb-raised-cards--uniform-height` to the Field Formatter Classes.
![image](https://user-images.githubusercontent.com/9355698/99319986-951e4280-2838-11eb-881d-6768656af377.png)
- Click update
- Click Save
- Check the page again and those classes should be applied around the container of the cards, `hb-raised-cards` is already applied to this page, so you don't have to apply it in the display.
4. To test this within Layout Builder, I found this easiest to create my own test page by adding a new "Flexible Page".
- Once I added the new page, I added a paragraph row with three cards.
- You can vary the content of each card, but try to leave one of the cards with very little text and no image. This will give you the ability to see whether the height and width are working or reliant on the content within the card to work.
- Both width and height should stay consistent regardless of content within each card.
- To add the classes you'll click the "Layout" tab at the top of the page content.
- Scroll down to the Components area of the layout builder layout and click on the contextual link for configure.
![image](https://user-images.githubusercontent.com/9355698/99320601-014d7600-283a-11eb-9e0a-65859587ff2c.png)
- Once there you can add the classes `hb-raised-cards` `hb-raised-cards--uniform-height` to the Field Formatter Class area.
- Click Update
- Click Save Layout
- Check your page again and the raised card classes, as well as the uniform modifier, should be active for verification.
![image](https://user-images.githubusercontent.com/9355698/99320538-dfec8a00-2839-11eb-83b9-e6994a1ae707.png)

5. This work should work the same on both Colorful and Traditional themes.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
